pkg/controllers/nodeclass/controller.go

```diff
@@ -80,6 +91,31 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeCl
 	return reconcile.Result{RequeueAfter: 5 * time.Minute}, err
 }
 
+func (c *Controller) Finalize(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
+	stored := nodeClass.DeepCopy()
+	if !controllerutil.ContainsFinalizer(nodeClass, v1beta1.TerminationFinalizer) {
+		return reconcile.Result{}, nil
+	}
+	ids, err := c.instanceProfileProvider.AssociatedInstances(ctx, nodeClass)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("terminating instance profile, %w", err)
+	}
+	if len(ids) > 0 {
+		c.recorder.Publish(WaitingOnInstanceTerminationEvent(nodeClass, instanceprofile.GetProfileName(ctx, nodeClass), ids))
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
I know this does a exponential backoff, but thoughts on just making this a `RequeueAfter: 10s` by default? Would prevent us from hammering the API calls, and since this is finalizing, we probably don't need to be super quick about this either.